File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.cpp
class Solution {
    int n ;
    vector<vector<int>> dp;
public:
    int rec(vector<int>&nums,int i,int previ){
        if(i==n) return 0;
        if(dp[i][previ+1]!=-1) return dp[i][previ+1];

        //loop over all choices 
        //skip
        int ans = rec(nums,i+1,previ);
        //take
        if(previ == -1 || nums[i]>nums[previ]) ans = max(ans, 1+rec(nums,i+1,i));
        return dp[i][previ+1]= ans;

    }
    int lengthOfLIS(vector<int>& nums) {
        n = nums.size();
        dp.assign(n+1,vector<int>(n+1,-1));
        return rec(nums,0,-1);
    }
};

File: Data Structures & Algorithms/counting-bits/submission-0.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        ans = []
        for i in range(0,n+1):
            cnt = 0
            while(i):
                i = i&(i-1)
                cnt = cnt+1
            ans.append(cnt)
        return ans

